### ISSUE TITLE
CO2 Analysis -- eGRID Subregion Code not picked up by program 

### ISSUE THREAD

**Issue by RDmitchell** [SEED-platform/seed#2997]
**Describe the bug**

- I imported a file with meter data (JCC anon) but without eGRID Subregion Code field
- Imported the same file again, but with that field added, and filled out
![image](https://user-images.githubusercontent.com/6314950/140582022-8904eb8c-42de-4576-a4d3-b99e8f2ef36c.png)
- Performed CO2 analysis, program says that eGRID subregion code is missing
![image](https://user-images.githubusercontent.com/6314950/140582005-bd722388-81d9-4a56-8c46-f15b3eba72b6.png)

**Expected Behavior**
I would have expected the program to calculate the CO2 values because I imported the eGRID subregion codes

**Actual Behavior**
see description

**Steps to Reproduce**
See description

**Instance Information**
Instance: ad707077bc25e403eafed1eb5cc6874d-8554735.us-east-1.elb.amazonaws.com
SHA: 6a31f3d36

**Comment by RDmitchell** [comment]
@macintoshpie -- I have added this to the current project, so it is in Triage

**Comment by RDmitchell** [comment]
@macintoshpie / @nllong -- copying what I said in slack into this issue

I think something might be wrong with the CO2 analysis on the staging server I am testing on. I thought it had to do with the fact that I added the eGRID subregion field and data on a 2nd import (see issue 2997) but I deleted the inventory (as a superuser) and then reimported the file with the egrid region specified, and am still getting the error that "Property is missing eGRID Subregion Code".
![image](https://user-images.githubusercontent.com/6314950/140585626-fa4bf738-3365-45a6-925b-59fffef289e0.png)
The egrid subregion code is defined for all the records
![image](https://user-images.githubusercontent.com/6314950/140585645-5fe21d85-2dbe-43ac-ad63-c80b02a87567.png)

**Comment by macintoshpie** [comment]
Not sure if this is causing the issue, but we're grabbing egrid subregion code from the PropertyView instead of the AnalysisPropertyView (which it should be the latter)
https://github.com/SEED-platform/seed/blob/6a31f3d369af1348afccb14c33ff9b4925cb711b/seed/analysis_pipelines/co2.py#L384

**Comment by RDmitchell** [comment]
@macintoshpie -- so does that mean you will need to change the code? If yes, let me know when the fix is deployed to the new staging server and I will retest. Thx.

**Comment by RDmitchell** [comment]
Here is the file I imported.

[JCC_ESPM_clean_anonymized-40goodrecordsGas-Elec-EGRID.xlsx](https://github.com/SEED-platform/seed/files/7489408/JCC_ESPM_clean_anonymized-40goodrecordsGas-Elec-EGRID.xlsx)

**Comment by RDmitchell** [comment]
@macintoshpie 
Here is the mapping
![image](https://user-images.githubusercontent.com/6314950/140591109-041f36ef-efdb-4888-8ff4-dc6316865830.png)

**Comment by macintoshpie** [comment]
@RDmitchell posted in slack, including here as well for documentation:

> My guess is when you tested on monday, the “blessed” column didn’t exist yet (it wasnt in 2.12.1), then in 2.12.2 we added the column,  and when you were reimporting it somehow ended up mapping to your existing extra data column

If you could check your column settings and verify there's no egrid code extra data col and retry that might fix it.
I tested with your provided data and it worked properly

**Comment by RDmitchell** [comment]
@macintoshpie -- I will try the import again, but I believe I did this test in this version 2.12.2.6a31f3d36

**Comment by RDmitchell** [comment]
@macintoshpie -- I suspect this is the problem, ie, I have two definitions of eGRID Subregion Code, one in extra data. 
![image](https://user-images.githubusercontent.com/6314950/140805980-aa637b13-6e7b-489e-8ca5-cbed93aedd64.png)

I guess I should delete the Extra Data field?

**Comment by RDmitchell** [comment]
To get the CO2 analysis to work, I tried the following

* I deleted the inventory records and the previous data files from the dataset
* when mapping the fields, when given the option of the two fields, I selected the "eGRID Subregion Code" that was highlighted in green, which I interpreted was the real field and not the extra data field. 
* That did not seem to allow the CO2 analysis to run
* I then deleted the "eGRID Subregion Code" that was in extra data in Column Settings. 
* I then deleted the inventory and the previous data set files
* I then reimported the JCC 40 records file, had only one option for mapping "eGRID Subregion Code", used that, and now the CO2 analysis is working.
